lib/index.js

```diff
@@ -118,14 +118,13 @@ internals.Client.prototype.request = function (method, url, options, callback, _
 
     uri.method = method.toUpperCase();
     uri.headers = options.headers || {};
-    const hasContentLength = Object.keys(uri.headers).some((key) => {
-
-        return key.toLowerCase() === 'content-length';
-    });
+    const hasContentLength = Boolean(internals.findHeader('content-length', uri.headers));
```

Comment:
Just to determine the cases where this should be `false` based on a value and not on its type.

At the end this variable answers the question "does it has content-length?" not "does it has a content-length which type is `number`?"